src/main/kotlin/com/ampnet/payoutservice/repository/JooqCreatePayoutTaskRepository.kt

```diff
@@ -0,0 +1,125 @@
+            requesterAddress = WalletAddress(task.requesterAddress!!),
+            issuerAddress = task.issuerAddress?.let { ContractAddress(it) },
+            data = taskData
+        )
+    }
+
+    override fun createPayoutTask(params: CreatePayoutTaskParams): UUID {
+        logger.info { "Storing create payout task, params: $params" }
+
+        val taskId = uuidProvider.getUuid()
+
+        dslContext.executeInsert(
+            CreatePayoutTaskRecord(
+                id = taskId,
+                chainId = params.chainId.value,
+                assetAddress = params.assetAddress.rawValue,
+                blockNumber = params.payoutBlock.value,
+                ignoredAssetAddresses = params.ignoredAssetAddresses.map { it.rawValue }.toTypedArray(),
+                requesterAddress = params.requesterAddress.rawValue,
+                issuerAddress = params.issuerAddress?.rawValue,
+                status = DbTaskStatus.PENDING,
+                resultTree = null,
+                treeIpfsHash = null,
+                totalAssetAmount = null
+            )
+        )
+
+        return taskId
+    }
+
+    override fun getPending(): PendingCreatePayoutTask? {
+        return dslContext.selectFrom(CreatePayoutTaskTable.CREATE_PAYOUT_TASK)
+            .where(CreatePayoutTaskTable.CREATE_PAYOUT_TASK.STATUS.eq(DbTaskStatus.PENDING))
+            .limit(1)
+            .forUpdate()
+            .skipLocked()
+            .fetchOne()?.let {
+                PendingCreatePayoutTask(
+                    taskId = it.id!!,
+                    chainId = ChainId(it.chainId!!),
+                    assetAddress = ContractAddress(it.assetAddress!!),
+                    blockNumber = BlockNumber(it.blockNumber!!),
+                    ignoredAssetAddresses = it.ignoredAssetAddresses!!.mapTo(HashSet()) { a -> WalletAddress(a!!) },
+                    requesterAddress = WalletAddress(it.requesterAddress!!),
+                    issuerAddress = it.issuerAddress?.let { i -> ContractAddress(i) }
+                )
+            }
+    }
+
+    override fun completeTask(
+        taskId: UUID,
+        merkleTreeRootId: UUID,
+        merkleTreeIpfsHash: IpfsHash,
+        totalAssetAmount: BigInteger
+    ) {
+        logger.info {
+            "Marking task as success, taskId: $taskId, merkleTreeRootId: $merkleTreeRootId," +
+                " merkleTreeIpfsHash: $merkleTreeIpfsHash, totalAssetAmount: $totalAssetAmount"
+        }
+        dslContext.update(CreatePayoutTaskTable.CREATE_PAYOUT_TASK)
+            .set(CreatePayoutTaskTable.CREATE_PAYOUT_TASK.STATUS, DbTaskStatus.SUCCESS)
+            .set(CreatePayoutTaskTable.CREATE_PAYOUT_TASK.RESULT_TREE, merkleTreeRootId)
+            .set(CreatePayoutTaskTable.CREATE_PAYOUT_TASK.TREE_IPFS_HASH, merkleTreeIpfsHash.value)
+            .set(CreatePayoutTaskTable.CREATE_PAYOUT_TASK.TOTAL_ASSET_AMOUNT, totalAssetAmount)
+            .execute()
```

Comment:
If this execute can return an updated record from the database, it would be nice to return it at the end of the function even if it is not used.
Nothing mandatory, just nice to have, the function that changes the value to return the updated value.